src/dnn/MIOpen.jl

```diff
@@ -0,0 +1,77 @@
+module MIOpen
+
+import AMDGPU
+import AMDGPU.Runtime.Mem
+
+using AMDGPU: ROCArray, ROCDevice
+using CEnum
+using GPUArrays
+using MIOpen_jll
+
+const libMIOpen_path = MIOpen_jll.libMIOpen_path
+
+include("low_level.jl")
+
+const HANDLE = Ref{miopenHandle_t}(C_NULL)
+
+function get_status_string(status)
+    if     status == miopenStatusSuccess return "Success"
+    elseif status == miopenStatusNotInitialized return "Not initialized"
+    elseif status == miopenStatusInvalidValue return "Invalid value"
+    elseif status == miopenStatusBadParm return "Bad parameter"
+    elseif status == miopenStatusAllocFailed return "Allocation failed"
+    elseif status == miopenStatusInternalError return "Internal error"
+    elseif status == miopenStatusNotImplemented return "Not implemented"
+    elseif status == miopenStatusUnknownError return "Unknown error"
+    elseif status == miopenStatusUnsupportedOp return "Unsupported operation"
+    elseif status == miopenStatusGpuOperationsSkipped return "GPU operations skipped" end
```

Comment:
It's usually my preference to put these mappings into a `Dict`, and then `get_status_string` can do a lookup from it. I know that this `if ... elseif` logic probably makes for faster code, but it's more verbose and doesn't allowing checking (without throwing and catching an error) if a certain status code has an associated status string.